Capstone/All Comments/DetailCommentCell.m

```diff
@@ -0,0 +1,108 @@
+            self.downvotes = self.commentFromParse[@"downvotedUsers"];
+            [self voteStatusChecker];
+        } else {
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
+}
+
+- (IBAction)didTapUpvote:(id)sender {
+    PFQuery *query = [PFQuery queryWithClassName:@"Comments"];
+    [query whereKey:@"objectId" equalTo:self.commentObjectId];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *comments, NSError *error) {
+        if (comments != nil) {
+            self.commentFromParse = [comments objectAtIndex:0];
+            [self.downvoteHeartButton setImage:[UIImage imageNamed:@"arrow.down.heart.png"]forState:UIControlStateNormal];
+            [self.upvoteHeartButton setImage:[UIImage imageNamed:@"arrow.up.heart.fill.png"]forState:UIControlStateNormal];
+            self.upvotes = self.commentFromParse[@"upvoted"];
+            [self.upvotes addObject:self.currentUser.username];
+            self.commentFromParse[@"upvoted"] = [NSArray arrayWithArray:self.upvotes];
+            self.downvotes = self.commentFromParse[@"downvotedUsers"];
+            if ([self.downvotes containsObject:self.currentUser.username]) {
+                [self.downvotes removeObject:self.currentUser.username];
+                self.commentFromParse[@"downvotedUsers"] = [NSArray arrayWithArray:self.downvotes];
+            }
+            self.commentFromParse.vote++;
+            [self.commentFromParse saveInBackground];
+            [self counterUpdateForLabel];
+        } else {
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
+}
+
+- (IBAction)didTapDownVote:(id)sender {
+    PFQuery *query = [PFQuery queryWithClassName:@"Comments"];
+    [query whereKey:@"objectId" equalTo:self.commentObjectId];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *comments, NSError *error) {
+        if (comments != nil) {
+            self.commentFromParse = [comments objectAtIndex:0];
+            [self.upvoteHeartButton setImage:[UIImage imageNamed:@"arrow.up.heart.png"]forState:UIControlStateNormal];
+            [self.downvoteHeartButton setImage:[UIImage imageNamed:@"arrow.down.heart.fill.png"]forState:UIControlStateNormal];
+            self.downvotes = self.commentFromParse[@"downvotedUsers"];
+            [self.downvotes addObject:self.currentUser.username];
+            self.commentFromParse[@"downvotedUsers"] = [NSArray arrayWithArray:self.downvotes];
+            self.upvotes = self.commentFromParse[@"upvoted"];
+            if ([self.upvotes containsObject:self.currentUser.username]) {
+                [self.upvotes removeObject:self.currentUser.username];
+                self.commentFromParse[@"upvoted"] = [NSArray arrayWithArray:self.upvotes];
+            }
+            self.commentFromParse.vote--;
+            [self counterUpdateForLabel];
+            [self.commentFromParse saveInBackground];
+        } else {
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
+}
```

Comment:
+1 to this - there seems to be a lot of overlap between these two functions so it would be a lot more efficient to combine them using a helper function and take in a bool param like Angela suggested!

Reply:
done